tests/test_batched_sgp4.py

```diff
@@ -44,6 +44,48 @@ def test_sgp4_batched(self):
         #we batch propagate all TLEs at all required times:
         out_batched=dsgp4.propagate_batch(tles_batch,tsinces_batch)
         self.assertTrue(np.allclose(out_non_batched.numpy(),out_batched.numpy()))
+        
+    def test_isimp_batched(self):
+        lines = file1.splitlines()
+        index = list(range(1,4,3))
+        tles_batch      = []
+        tsinces_batch    = []
+        out_non_batched = []
+        for i in index:
+            data = []
+            data.append(lines[i])
+            data.append(lines[i+1])
+            data.append(lines[i+2])
+            tle = dsgp4.tle.TLE(data)
+            try: 
+                dsgp4.initialize_tle(tle)
+                if tle._error==0:
+                    tsince = torch.rand(100)*10
+                    tles_batch += [tle]*len(tsince)
+                    tsinces_batch+=[tsince]
+                    out_non_batched+=[dsgp4.propagate(tle,tsince)]
+            except Exception as e:
+                self.assertTrue(str(e).split()==error_string.split())
+        tsinces_batch = torch.cat(tsinces_batch)
+        out_non_batched = torch.cat(out_non_batched)
+        out_batched = dsgp4.propagate_batch(tles_batch,tsinces_batch)
+        self.assertTrue(np.allclose(out_non_batched.numpy(),out_batched.numpy()))
+
```

Comment:
can you please also add an assert here confirming that all `isimp` values are equal to 1? Otherwise this test is expected to pass anyway for `isimp==0`

Reply:
Yes, this test is expected to pass for isimp equal to 0 or equal to 1. The TLEs of file1 are both 0 and 1. In this way, the test proves that the results are the same for both cases.

Reply:
@Rex2000viola I see, but it is not clear that there is at least one TLE with `isimp`==1, so there should be an extra test to verify that at least one is `isimp==1`